extensions/data-workspace/src/main.ts

```diff
@@ -15,8 +15,12 @@ import { IconPathHelper } from './common/iconHelper';
 import { ProjectDashboard } from './dialogs/projectDashboard';
 import { getAzdataApi } from './common/utils';
 import { createNewProjectWithQuickpick } from './dialogs/newProjectQuickpick';
+import Logger from './common/logger';
 
 export async function activate(context: vscode.ExtensionContext): Promise<IExtension> {
+	Logger.log(`Starting Data Workspace activate()`);
```

Comment:
You should also be logging each separate action being done here. Because this is where the problem is - in the issue linked it's taking 13seconds for activate to resolve. So something in here is getting blocked/delayed for some reason Although I can't tell at first glance what that might be - these all should be quick calls...although it's possible something else is getting stuck and blocking the calls here from completing.

One thing to ask people too is if they see this consistently - have them start VS Code a couple of times and measure the activation time for the extension each time to see if it's just something that happens occasionally or something in their specific environment that's causing issues.

Reply:
Added more logging in `activate()`. I had just assumed that the long times would be from loading the projects or looking for project providers in extensions, but good to add more logging to be sure.